src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -610,6 +612,9 @@ private static bool StringRequiresHtmlDecoding(string s)
         // Internal struct to facilitate URL decoding -- keeps char buffer and byte buffer, allows appending of either chars or bytes
         private struct UrlDecoder
         {
+            public bool _containsUnsafe;
+            public bool _containsSpaces;
```

Comment:
Why store these as fields on the UrlDecoder?  Seems like they could just be a single local ```needsDecoding```, and then instead of doing ```return helper.GetString();```, you'd do ```return needsDecoding ? helper.GetString() : value;```.